swap_meet/vendor.py

```diff
@@ -1,2 +1,99 @@
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__ (self, inventory = None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        
+        self.inventory.remove(item)
+        return item
+    
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None    
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in other_vendor.inventory:
```

Comment:
When we have longer compound conditionals like this one, it can be a good idea to split individual conditionals across lines. We can do this using a backslash like the following:

```suggestion
        if my_item in self.inventory \
             and their_item in other_vendor.inventory:
```